### PR TITLE
Add plugin to parse values from log file

### PR DESCRIPTION
**Pull request by v0devil**

### CODE REVIEW

File: src/logfile_values.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('--prefix', default='logfile_values')
+    parser.add_argument('--file', default='1.log')
+    parser.add_argument(
+        '-n',
+        '--name',
+        help='Graphite metric name',
+        default=None,
```
Comment (hasegeli) [inline src/logfile_values.py]: This doesn't look useful.  The default would be `[]` when `action='append'`.

File: src/logfile_values.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+"""
+
+from argparse import ArgumentParser
+from time import time
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('--prefix', default='logfile_values')
+    parser.add_argument('--file', default='1.log')
```
Comment (hasegeli) [inline src/logfile_values.py]: This is a strange default.  How about something like `/var/log/messages`?

File: src/logfile_values.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    parser.add_argument(
+        '-n',
```
Comment (hasegeli) [inline src/logfile_values.py]: Do we really need the short variants?

File: src/logfile_values.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    parser.add_argument(
+        '-c',
+        '--column',
+        help='Number of the column with metric value',
+        default=None,
+        action='append',
+        dest='columns', )
+    args = parser.parse_args()
+    if len([x for x in (args.names) if x is not None]) != len(
+        [x for x in (args.columns) if x is not None]):
```
Comment (hasegeli) [inline src/logfile_values.py]: This would be `len(args.name) == len(args.columns)` without the `default`s.

File: src/logfile_values.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        default=None,
+        action='append',
+        dest='names', )
+    parser.add_argument(
+        '-c',
+        '--column',
+        help='Number of the column with metric value',
+        default=None,
+        action='append',
+        dest='columns', )
```
Comment (hasegeli) [inline src/logfile_values.py]: You can add there `type=int` to avoid casting later.

File: src/logfile_values.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    args = parser.parse_args()
```
Comment (hasegeli) [inline src/logfile_values.py]: I don't like dependent command like arguments.  I would design it something like this:

```python
class Metric:
    def __init__(self, arg):
        if ':' not in arg:
            raise ArgumentTypeError('Argument must have ":"')
        self.name, column = arg.split(':', 1)
        if not column.isdecimal():
            raise ArgumentTypeError('Column must be a number')
        self.column = int(column)
```

```python
    parser.add_argument('metric', type=Metric, action='append')
```

`ArgumentParser` lets you use simple functions as types as well, but I think class is better for this script.  We can extend it later to get the metrics from lines other than the last one.

File: src/logfile_values.py (hunk 4 above)
Comment (TheFRedFox) [reply src/logfile_values.py]: Generally `[x for x in (args.names) if x is not None]` can be replaced with:
`filter(None, args.names)`

**Pull request merged**